cardano-testnet/src/Testnet/Runtime.hs

```diff
@@ -290,29 +289,28 @@ startLedgerNewEpochStateLogging testnetRuntime tmpWorkspace = withFrozenCallStac
       logFile = logDir </> "ledger-epoch-state.log"
       diffFile = logDir </> "ledger-epoch-state-diffs.log"
 
-  H.evalIO (IO.doesDirectoryExist logDir) >>= \case
+  liftIOAnnotated $ IO.doesDirectoryExist logDir >>= \case
     True -> pure ()
     False -> do
-      H.note_ $ "Log directory does not exist: " <> logDir <> " - cannot start logging epoch states"
-      H.failure
-
-  H.evalIO (IO.doesFileExist logFile) >>= \case
-    True -> do
-      H.note_ $ "Epoch states logging to " <> logFile <> " is already started."
-    False -> do
-      H.evalIO $ appendFile logFile ""
-      socketPath <- H.noteM $ H.sprocketSystemName <$> H.headM (testnetSprockets testnetRuntime)
-
-      _ <- H.asyncRegister_ . runExceptT $
-        foldEpochState
-          (configurationFile testnetRuntime)
-          (Api.File socketPath)
-          Api.QuickValidation
-          (EpochNo maxBound)
-          Nothing
-          (handler logFile diffFile)
-
-      H.note_ $ "Started logging epoch states to: " <> logFile <> "\nEpoch state diffs are logged to: " <> diffFile
+      throwString $ "Log directory does not exist: " <> logDir <> " - cannot start logging epoch states"
+
+  liftIOAnnotated $ IO.doesFileExist logFile >>= \case
+    True -> return () 
+    False -> liftIO $ appendFile logFile ""
+
+  let socketPath = case uncons (testnetSprockets testnetRuntime) of
+        Just (sprocket, _) -> H.sprocketSystemName sprocket
+        Nothing            -> throwString "No testnet sprocket available"
+
+  let act = runExceptT $
+              foldEpochState
+                (configurationFile testnetRuntime)
+                (Api.File socketPath)
+                Api.QuickValidation
+                (EpochNo maxBound)
+                Nothing
+                (handler logFile diffFile)
+  void $ asyncRegister_ act
```

Comment:
This part of the code has changed. Previously it was only logging if the output file did not exist already. Now it will mangle the output of multiple epoch state loggers if they get enabled at the same time.

This code should be inside `False` branch of the `case ` in line 299.

Reply:
Good catch :+1: